pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -171,9 +171,7 @@ func (rc *reconciler) reconcile() {
 			// See https://github.com/kubernetes/kubernetes/issues/93902
 			attachState := rc.actualStateOfWorld.GetAttachState(attachedVolume.VolumeName, attachedVolume.NodeName)
 			if attachState == cache.AttachStateDetached {
-				if klog.V(5).Enabled() {
-					klog.Infof(attachedVolume.GenerateMsgDetailed("Volume detached--skipping", ""))
-				}
+				klog.V(5).InfoS("Volume detached--skipping", "volume", attachedVolume)
```

Comment:
how does this compare to the previous output of GenerateMsgDetailed? this PR would be easier to review if it was a mechanical translation of the verbosity check.

Reply:
(looks like tim/dims already acked, this comment isn't blocking, I just wondered)

Reply:
The output is now
```
"Volume detached--skipping" volume="<details about volume>"
```
instead of a similar string with the same information.

The conversion to structured logging has the advantage that the (kind of) expensive formatting of the volume object is only done if needed, even without the explicit if check. As you said, it's a larger change, but it is the right solution for this problem here.